src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java

```diff
@@ -578,13 +581,12 @@ public long handleEvent(final int event, long data1, long data2) {
         break;
 
       case DC_EVENT_INCOMING_MSG:
-        MessageNotifierCompat.updateNotification((int) data1, (int) data2); // updateNotification() makes sure to run in the correct thread
+        notificationCenter.addNotification((int) data1, (int) data2);
```

Comment:
I think it would be cleaner to handle the notification here as an EventObserver.
Currently we have this one event and event handler that are outside the existing Event segment. That would need a good explanation

Reply:
well, we just have not "global" event handler, and i did not want to change this in this pr. apart from that, i think, it is fine - we also handle errors here, and we all the things in a file that also starts threads etc. not sure, an additional indirection helps here for better overview or so.